tools/fw-update/rs-fw-update.cpp

```diff
@@ -175,33 +169,35 @@ void waiting_for_device_to_reconnect(rs2::context& ctx, rs2::cli::value<std::str
 
 }
 
+bool is_fw_compatible(const rs2::device& dev, const std::vector< uint8_t >& fw_image)
```

Comment:
This function name is `is_fw_compatible`, the misusage in D555 is wrong, but for others lets keep it simple, it should check if the FW is compatible for the device.
There are several use cases both on USB MIPI and DDS that you should check if you refactor this tool.
Operational device flow
Recovery device flow
For all connection types